File: day2/p1/solution.rb
def checksum(matrix)
  matrix.reduce(0) do |agg, arr|
    min, max = arr.minmax # Ruby is great
    agg + (max-min)
  end
end

matrix = File.read('input.txt').split("\n").map { |r| r.split("\s").map(&:to_i) }
puts checksum(matrix)

File: day2/p1/solution.exs
defmodule D2P1 do
  def checksum(matrix) do
    matrix
    |>  Enum.reduce(0, fn(arr, agg) ->
      {min, max} = Enum.min_max(arr)
      agg + (max - min)
    end)
  end
end

{:ok, input} = File.read("input.txt")

matrix = input
|> String.split("\n", trim: true)
|> Enum.map(fn(str) ->
  str
  |> String.split(~r/\s+/, trim: true)
  |> Enum.map(fn(num) -> String.to_integer(num) end)
end)

IO.inspect(D2P1.checksum(matrix))
